### DISCUSSION TITLE
LongPress duration

### DISCUSSION THREAD

**Discussion by KrSerg** [react-native-tvos/react-native-tvos#778] [General Discussions]
I wanted to ask about the possibility of setting a custom value for handling a LongPress event, but I found out that these values are fixed: 300ms for Android and the standard 500ms for tvOS—please correct me if I'm wrong.

This brings me to my next question: Are there any plans to add the ability to modify these values? 
Firstly, to make them consistent across platforms, and secondly, to increase the value without needing additional logic, such as timers. Reducing the value doesn’t seem feasible, as tvOS lacks a keyDown event, making it impossible to handle this case.

**Comment by kosso** [comment]
So, the `TVEventHandler` has `{ eventType, eventKeyAction }` in the event. 

`eventKeyAction = 0` should be keyDown and `eventKeyAction = 1` should be keyUp.

You're right that the keyDown (0) never normally occurs on tvOS. But I've just discovered that it does fire `eventKeyAction = 0` when the `eventType` is `longUp`, `longDown`, `longLeft` or `longRight` is triggered.

I was recently in the process of building a hook to detect long-presses, with a variable delay, just using the `up` and `down` eventTypes when I discovered the built-in `long**` types.

**Reply by kosso** [reply]
I know it doesn't deal with the delay requirement, but this is what I ended up with 

```
export const useTVRemoteLongPress = (onLongPressed: any) => {
  const subscriptionRef = useRef(null)
  useEffect(() => {
    const handleTVEvent = (event: any) => {
      const { eventType, eventKeyAction } = event;
      if (eventType === 'longUp' || eventType === 'longDown' || eventType === 'longLeft' || eventType === 'longRight') {
        if (eventKeyAction === 0) {
          onLongPressed(true, eventType)
        } else if (eventKeyAction === 1) {
          onLongPressed(false, eventType)
        }
      }
    };
    subscriptionRef.current = TVEventHandler.addListener(handleTVEvent)
    return () => {
      subscriptionRef.current?.remove();
    }
  }, [onLongPressed])
}
```

Usage:
```
useTVRemoteLongPress((longPressed: boolean, type: string) => {
    console.log(`longPress: ${longPressed ? 'keydown' : 'keyup'} - type: ${type}`)
    // do things ...
})
```
